### ISSUE TITLE
Non-compliance to ARIA and WCAG in Forum

### ISSUE THREAD

**Issue by umitcan07** [bounswe/bounswe2024group11#603]
# 🧱 Description

Some of the components in forums page and forum inner page do not compy with our guidelines about WCAG and WAI-ARIA.

- Upvote Downvote buttons have no label.
- Forum card inside forums page is not a proper link and has no description.
- Author has no description.
- 
## 🛣️ Reproduction

- Open forums page. There will be missing attributes.
- Open a forum page. There will be missing attributes.

## 📋 Acceptance Criteria

- [ ] Upvote Downvote buttons have aria-label or aria-labelled-by
- [ ] Forum card is a proper link with href (in the resultant markup, not necessarily source code, so we can use <Link> from react-router)
- [ ] Forum card has aria-label or aria-description with the forum title and author.
